datadog-setup.php

```diff
@@ -1444,12 +1492,18 @@ function download($url, $destination)
         curl_setopt($ch, CURLOPT_NOPROGRESS, false);
         $progress_counter = 0;
         $return = curl_exec($ch);
+
+        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        if ($httpCode == 404) {
```

Comment:
This looks weird to me.
IIUC, at the first download attempt (for a specific version), if `curl` is installed, we are going to try only with curl.
Then at the second attempt (full package), this could create an infinite loop if github (or a proxy or whatever) returns a 404.

Reply:
You're right, I messed that up.

Reply:
Wrapped in $retry, looks good now?